crates/assets/js/client/src/index.ts

```diff
@@ -275,71 +375,110 @@ export class RecordApiImpl<T = Record<string, unknown>>
       }
     }
 
-    const filters = opts?.filters;
+    const filters = this.opts?.filters;
     if (filters) {
       for (const filter of filters) {
         traverseFilters("filter", filter);
       }
     }
 
-    const response = await this.client.fetch(`${this._path}?${params}`);
+    const response = await this.client.fetch(
+      `${recordApiBasePath}/${this.apiName}?${params}`,
+    );
     return (await response.json()) as ListResponse<T>;
   }
+}
 
-  public async read<T = Record<string, unknown>>(
+export interface RecordApi<T = Record<string, unknown>> {
+  // Immediate operations
+  list(opts?: {
+    pagination?: Pagination;
+    order?: string[];
+    filters?: FilterOrComposite[];
+    count?: boolean;
+    expand?: string[];
+  }): Promise<ListResponse<T>>;
+
+  read(
     id: string | number,
     opt?: {
       expand?: string[];
     },
-  ): Promise<T> {
-    const expand = opt?.expand;
-    const response = await this.client.fetch(
-      expand
-        ? `${this._path}/${id}?expand=${expand.join(",")}`
-        : `${this._path}/${id}`,
-    );
-    return (await response.json()) as T;
-  }
+  ): Promise<T>;
 
-  public async create<T = Record<string, unknown>>(
-    record: T,
-  ): Promise<string | number> {
-    const response = await this.client.fetch(this._path, {
-      method: "POST",
-      body: JSON.stringify(record),
-      headers: jsonContentTypeHeader,
-    });
+  create(record: T): Promise<string | number>;
 
-    return (await response.json()).ids[0];
-  }
+  update(id: string | number, record: Partial<T>): Promise<void>;
 
-  public async createBulk<T = Record<string, unknown>>(
-    records: T[],
-  ): Promise<(string | number)[]> {
-    const response = await this.client.fetch(this._path, {
-      method: "POST",
-      body: JSON.stringify(records),
-      headers: jsonContentTypeHeader,
-    });
+  delete(id: string | number): Promise<void>;
 
-    return (await response.json()).ids;
+  subscribe(id: string | number): Promise<ReadableStream<Event>>;
+
+  // Deferred operations
+  listOp(opts?: {
+    pagination?: Pagination;
+    order?: string[];
+    filters?: FilterOrComposite[];
+    count?: boolean;
+    expand?: string[];
+  }): ListOperation<T>;
+
+  readOp(
```

Comment:
:super nitty: just from a discoverability point of view, i'd group <operation> and <operation>Op next to each other.